NbMGui/src/main/java/nbmgui/Controller.java

```diff
@@ -76,6 +83,44 @@ void openFile(ActionEvent event) {
             outputView.appendText("Select a NoBeard object file\n");
     }
 
+    @FXML
+    void startProgram(ActionEvent event) {
+        setBreakpoints();
+        stepButton.setDisable(false);
+        continueButton.setDisable(false);
+        stopButton.setDisable(false);
+        Thread thread = new Thread(new Runnable() {
+            @Override
+            public void run() {
+                machine.loadStringConstants(objectFile.getStringStorage());
+                machine.loadProgram(0, objectFile.getProgram());
+                machine.runProgram(0);
+                setCurrentPcOfUI();
+            }
+        });
+        thread.start();
+    }
+
+    @FXML
+    void step(ActionEvent event) {
+        machine.step();
```

Comment:
ah yes. Understood. Nevertheless I would suggest to go towards the solution I proposed (Change the breakpoints on click on the checkbox)